Bottom: ed3a8cf082d2e65dd5ffd3d33a488dd87470ee2a
Top:    b7de5ca8b0326ee7422505b5ac00f365e53e8286
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-26 03:11:21 +0100

Refresh of play-with-perspective-renderin

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index c71c430..a78c5fb 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -914,13 +914,26 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
 
   glMatrixMode (GL_PROJECTION);
   glLoadIdentity ();
-  glOrtho (0, widget->allocation.width, widget->allocation.height, 0, -100000, 100000);
+//  glOrtho (0, widget->allocation.width, widget->allocation.height, 0, -100000, 100000);
+//  glOrtho (-1, 1, -1, 1, 100, 10000);
+//  glFrustum (0, widget->allocation.width, widget->allocation.height, 0, 10, 1000);
+  glFrustum (-1, 1, -1, 1, 10, 10000);
+//  gluPerspective (30, widget->allocation.width / widget->allocation.height, 10, 1000);
   glMatrixMode (GL_MODELVIEW);
   glLoadIdentity ();
 
+  glScalef (2, 2, 1);
+
+  glTranslatef (-1., 1., 0.);
+  glScalef ( 2. / (float)widget->allocation.width,
+            -2. / (float)widget->allocation.height,
+             1. / 100.);
+
+  glTranslatef (0., 0., -2000.);
   glTranslatef (widget->allocation.width / 2., widget->allocation.height / 2., 0);
   glMultMatrixf ((GLfloat *)view_matrix);
   glTranslatef (-widget->allocation.width / 2., -widget->allocation.height / 2., 0);
+  glScalef (1., 1., 1. / 3.);
   glGetFloatv (GL_MODELVIEW_MATRIX, (GLfloat *)last_modelview_matrix);
 
   glEnable (GL_STENCIL_TEST);
